### PR TITLE
Backport `Eq`/`Ord` instances for `Generically1`

### LINKED ISSUE #2: Backport `Eq`/`Ord` instances for `Generically` introduced in `base-4.18`
_(Spun off from a discussion in https://github.com/haskell-compat/base-orphans/issues/63#issuecomment-1210476826.)_

`base-4.18.0.0` (bundled with GHC 9.6.1) introduces `Eq` and `Ord` instances for `Generically1` in [this commit](https://gitlab.haskell.org/ghc/ghc/-/commit/c4c2cca012c139259ab40e5b2e5f43aafc8f49c0). I plan to backport these instances as far back as `base` can support in haskell-compat/base-orphans#64, but we should also backport these instances to the `generically` library for even greater compatibility.

### PR DESCRIPTION
**Pull request by RyanGlScott**
Fixes #2.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+# next -- ????.??.??
+
+* Backport `Eq` and `Ord` instances for `Generically1`, which were introduced
+  in `base-4.18.0.0`.
```
Comment (phadej) [inline CHANGELOG.md]: I'd add add dependency to `base-orphans` with these instances to always "pollute" / provide them from `GHC.Generics.Generically`.

Comment (RyanGlScott) [reply CHANGELOG.md]: Good idea. In that case, I will park this PR for now and update it after haskell-compat/base-orphans#65 has been released to Hackage.

Comment (RyanGlScott) [reply CHANGELOG.md]: I've updated the PR to bring `Data.Orphans` into scope when needed.

**Comment by RyanGlScott** [comment]
If you don't mind, I'll plan on cutting a new Hackage release of `generically` within the next couple of days containing these changes.

**Comment by RyanGlScott** [comment]
I've uploaded [`generically-0.1.1`](https://hackage.haskell.org/package/generically-0.1.1) to Hackage with these changes.

**Pull request merged**